### PR TITLE
Feature/subsequences

### PR DESCRIPTION
**Pull request by WilliamHPNielsen**
Add support for subsequences in broadbean.

Adding support for subsequences turned out to necessitate (or at least strongly encourage) a refactoring of some of the underlying data juggling. This PR will aim to do that.

Specifically, a standard format for a *forged sequence* has been agreed upon. Everything else (drivers that output specific formats, plotting) will be recast to use the forged sequences.

I will flag it here when the PR is ready for review: 
- [x] Review time!

### CODE REVIEW

**Comment by WilliamHPNielsen** [comment]
Okay, I think this is finally in place. @QCoDeS/core

File: broadbean/broadbean.py [hunk 1]
```diff
@@ -1812,20 +1955,28 @@ def _plotSequence(self, elements):
...
+                    ax.annotate('SUBSEQ', xy=(0.5, 0.5),
+                                xycoords='axes fraction',
+                                horizontalalignment='center')
+                    time = np.linspace(0, 1, 2)  # needed for timeexponent
+
                 # Figure out the axes' scaling
                 timeexponent = np.log10(time.max())
                 timeunit = 's'
-                timescaling = 1
+                timescaling: float = 1
```
Comment (jenshnielsen) [inline broadbean/broadbean.py]: Make this really a float? i.e. 1.0

File: broadbean/broadbean.py [hunk 2]
```diff
@@ -1907,6 +2077,90 @@ def _plotSequence(self, elements):
...
+
+        output: Dict[int, Dict] = {}
+        channels = self.channels
+        data = deepcopy(self._data)
+        seqlen = len(data.keys())
+
+        # TODO: in this function, we iterate through the sequence three times
+        # It is probably worth considering refactoring that into a single
+        # iteration, although that may compromise readability
+
```
Comment (jenshnielsen) [inline broadbean/broadbean.py]: Alternatively perhaps the code below should be split into multiple functions

File: broadbean/broadbean.py [hunk 3]
```diff
@@ -1185,6 +1196,60 @@ def changeDuration(self, channel, name, newdur, replaceeverywhere=False):
 
         bp.changeDuration(name, newdur, replaceeverywhere)
 
+    def _applyDelays(self, delays: List[float]) -> None:
```
[the remark this answers is not in the document]
Comment (WilliamHPNielsen) [reply broadbean/broadbean.py]: I think that it's not worth it. The Sequence name is also taken already in this module.

**Pull request merged**